core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -63,11 +68,11 @@
   private final SerializingExecutor callExecutor;
   private final boolean unaryRequest;
   private final CallOptions callOptions;
-  private ClientStream stream;
+  private DelayedStream stream;
```

Comment:
A volatile read is indeed very cheap on x86, but on Android it is approximately the same cost as a lock (but without the concern of blocking). In addition, all the `set*`s are synchronized. All that is "okay," but it would be beneficial not to have it on the fast-path so that we aren't forced to reduce readability in order to gain performance.

If it _needs_ to be on the fast-path, then it would need to be on the fast-path and we have to live with the consequences (and possibly redesign things so it wasn't necessary). But since having it on the fast-path doesn't seem required, avoiding it now seems best.

From what I've read, avoiding it on the fast-path should be about as readable/maintainable as the code is now. If that's not the case, then we could re-evaluate.